controllers/admin/AdminModulesController.php

```diff
@@ -288,883 +91,76 @@ protected function getCurrentUrl($remove = [])
-                    if (Tools::getValue('enable')) {
-                        $module->enable();
-                    } else {
-                        $module->disable();
-                    }
-                    Tools::redirectAdmin($this->getCurrentUrl('enable'));
-                }
-            } else {
-                $this->errors[] = $this->trans('Cannot load the module\'s object.', [], 'Admin.Modules.Notification');
-            }
-        } else {
-            $this->errors[] = $this->trans('You do not have permission to add this.', [], 'Admin.Notifications.Error');
+        // only accept configuring a module
+        if (Tools::getIsset('configure')) {
+            return true;
         }
-    }
 
-    public function postProcessEnable_Device()
-    {
-        if ($this->access('edit')) {
-            $module = Module::getInstanceByName(Tools::getValue('module_name'));
-            if (Validate::isLoadedObject($module)) {
-                if (!$module->getPermission('configure')) {
-                    $this->errors[] = $this->trans('You do not have the permission to use this module.', [], 'Admin.Modules.Notification');
-                } else {
-                    $module->enableDevice((int) Tools::getValue('enable_device'));
-                    Tools::redirectAdmin($this->getCurrentUrl('enable_device'));
-                }
-            } else {
-                $this->errors[] = $this->trans('Cannot load the module\'s object.', [], 'Admin.Modules.Notification');
-            }
-        } else {
-            $this->errors[] = $this->trans('You do not have permission to add this.', [], 'Admin.Notifications.Error');
-        }
-    }
+        // redirect to module manager
+        Tools::redirectAdmin($this->context->link->getAdminLink('AdminModulesSf'));
 
-    public function postProcessDisable_Device()
-    {
-        if ($this->access('edit')) {
-            $module = Module::getInstanceByName(Tools::getValue('module_name'));
-            if (Validate::isLoadedObject($module)) {
-                if (!$module->getPermission('configure')) {
-                    $this->errors[] = $this->trans('You do not have the permission to use this module.', [], 'Admin.Modules.Notification');
-                } else {
-                    $module->disableDevice((int) Tools::getValue('disable_device'));
-                    Tools::redirectAdmin($this->getCurrentUrl('disable_device'));
-                }
-            } else {
-                $this->errors[] = $this->trans('Cannot load the module\'s object.', [], 'Admin.Modules.Notification');
-            }
-        } else {
-            $this->errors[] = $this->trans('You do not have permission to add this.', [], 'Admin.Notifications.Error');
-        }
+        return false;
     }
 
-    public function postProcessDelete()
+    /**
+     * {@inheritdoc}
+     */
+    public function postProcess()
     {
         /* PrestaShop demo mode */
+        /* @phpstan-ignore-next-line */
```

Comment:
You can add `_PS_MODE_DEMO_` as `dynamicConstantNames` in phpstan config (https://phpstan.org/config-reference#constants)

Reply:
Merge error, this should have been removed
```suggestion
```